lib/instrumentation/modules/elasticsearch.js

```diff
@@ -1,9 +1,58 @@
 'use strict'
 
+const URL = require('url').URL
+
 var shimmer = require('../shimmer')
 var { getDBDestination } = require('../context')
 const { setElasticsearchDbContext } = require('../elasticsearch-shared')
 
+const startsWithProtocolRE = /^([a-z]+:)?\/\//
```

Comment:
Does this regular expression need an `/.../i` modifier, or will _other code_ have normalized protocals with weird casing (HTTP://example.com`, etc.)

Reply:
I "stole" it from here: https://github.com/elastic/elasticsearch-js-legacy/blob/16.x/src/lib/host.js

My read of https://github.com/elastic/elasticsearch-js-legacy/blob/16.x/src/lib/host.js#L77-L100 is that given `new Client(host: 'HTTP://example.com')` that the legacy ES client will result in:

```
> new URL("http://HTTP://localhost")
URL {
  href: 'http://http//localhost',
  origin: 'http://http',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'http',
  hostname: 'http',
  port: '',
  pathname: '//localhost',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

I think there may be subtle edge cases old ES client itself here.

I don't object to case-insensitive here. Or I could just add a comment to that `const startsWithProtocolRE` that this is taken from the ES client code itself... and let us cross this road if we get a bug report on it.  Preference?

Reply:
I added the 'i' flag.